lib/logstash/inputs/redis.rb

```diff
@@ -165,21 +167,11 @@ def connect
   # private
   def load_batch_script(redis)
     #A Redis Lua EVAL script to fetch a count of keys
-    #in case count is bigger than current items in queue whole queue will be returned without extra nil values
-    redis_script = <<EOF
-          local i = tonumber(ARGV[1])
-          local res = {}
-          local length = redis.call('llen',KEYS[1])
-          if length < i then i = length end
-          while (i > 0) do
-            local item = redis.call("lpop", KEYS[1])
-            if (not item) then
-              break
-            end
-            table.insert(res, item)
-            i = i-1
-          end
-          return res
+      redis_script = <<EOF
+        local batchsize = tonumber(ARGV[1])
+        local result = redis.call('lrange', KEYS[1], 0, batchsize)
+        redis.call('ltrim', KEYS[1], batchsize + 1, -1)
```

Comment:
My take on this:
- I don't think the script is a bad thing as such - unless a customer does not want us to put a LUA script on their server.
- The [script is a transaction](http://redis.io/topics/transactions#redis-scripting-and-transactions) and transactions may be removed.
- By using pipeline and multi in the ruby driver, we open ourselves up to a new code execution path that we have not debugged yet.  Maybe the ruby driver does it well but we don't need the agro.
- The evalsha method, wrt ruby/jruby, is lightweight.